tests/components/grid.test.ts

```diff
@@ -1427,6 +1428,34 @@ describe("Copy paste keyboard shortcut", () => {
     expect(getCellContent(model, "A2")).toEqual("things");
   });
 
+  test("cut zone gets cleared on paste if content/style is altered after cut", async () => {
+    setCellContent(model, "A1", "things");
+    setStyle(model, "A1", { bold: true });
+    selectCell(model, "A1");
+    document.body.dispatchEvent(getClipboardEvent("cut", clipboardData));
+    setCellContent(model, "A1", "new content");
+    setStyle(model, "A1", { bold: false });
+    selectCell(model, "A2");
+    document.body.dispatchEvent(getClipboardEvent("paste", clipboardData));
+    expect(getCellContent(model, "A2")).toEqual("things");
+    expect(getStyle(model, "A2")).toEqual({ bold: true });
+    expect(getCell(model, "A1")).toBe(undefined);
+  });
+
+  test("Cut and paste of a date format cell, by toggling formula visibility should return date", async () => {
+    model.dispatch("SET_FORMULA_VISIBILITY", { show: true });
+    setCellContent(model, "A1", "1");
+    setCellFormat(model, "A1", "m/d/yyyy");
+    document.body.dispatchEvent(getClipboardEvent("cut", clipboardData));
+    model.dispatch("SET_FORMULA_VISIBILITY", { show: false });
```

Comment:
you can definitely check the content of clipboardDatza, which is what we really are interested in